### DISCUSSION TITLE
DB shard corrupted on single node corrective action?

### DISCUSSION THREAD

**Discussion by bdoyle0182** [apache/couchdb#3305] [Q&A] [answered]
Hi I couldn't find anything in the docs on this so figured I'd ask here. We have a single shard corrupted for a single db (one file) on a single host of our three node cluster. I'm trying to figure out how we can fix this without recreating the db cluster wide. If I delete the db file on a single host, will internal replication within the cluster handle it and replicate the db back to the node?

**Comment by kocolosk** [comment, marked as answer]
100% yes that's a safe thing to do assuming your database is running with multiple shard replicas and something I've seen done many times successfully. A couple of things to note:

1. You can confirm the sharding of the database in question via the _shards endpoint: https://docs.couchdb.org/en/stable/api/database/shard.html 
2. Consumers of the `_changes` feed will automatically switch over to a different shard replica, which may cause some updates to be replayed on the feed.
3. You may want to put the node hosting the rebuilding shard file into "maintenance mode" to prevent it from contributing to responses before internal replication gets it caught up: https://docs.couchdb.org/en/stable/config/couchdb.html#couchdb/maintenance_mode

**Reply by bdoyle0182** [reply]
Thanks I had gone back and looked at some other threads and see that others had done it so I went ahead and tried it. It worked very smoothly
